crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1675,9 +1682,12 @@ where
                                 let concise_name = name.concise();
                                 debug!(?tx_digest, name=?name.concise(), weight, "Failed to get signed effects from validator handle_certificate: {:?}", err);
                                 self.metrics.process_cert_errors.with_label_values(&[&concise_name.to_string(), err.as_ref()]).inc();
+                                if let SuiError::RpcError(..) = &err {
+                                    state.bad_stake_rpc_error += weight;
+                                }
                                 state.errors.push((err, vec![name], weight));
                                 state.bad_stake += weight;
-                                if state.bad_stake > validity {
+                                if state.bad_stake - state.bad_stake_rpc_error > validity {
```

Comment:
doesn't this mean we will fail to terminate if > f validators are down? (until the outer timeout is reached).

I think we need to account for "in flight stake" and exit if there is not enough in-flight stake remaining to possibly push bad_stake high enough to clear this threshold.

Reply:
> I think we need to account for "in flight stake" and exit if there is not enough in-flight stake remaining to possibly push bad_stake high enough to clear this threshold.

this is a good point, will address